tutor/env.py

```diff
@@ -215,6 +215,69 @@ def __render(self, template: jinja2.Template) -> str:
             raise exceptions.TutorError(f"Missing configuration value: {e.args[0]}")
 
 
+class PatchRenderer(Renderer):
+    """
+    Render patches for print it.
+    """
+
+    def __init__(self, config: t.Optional[Config] = None):
+        self.patches_locations: t.Dict[str, t.List[str]] = {}
+        self.current_template: str = ""
+        super().__init__(config)
+
+    def render_template(self, template_name: str) -> t.Union[str, bytes]:
+        """
+        Set the current template and render template from Renderer.
+        """
+        self.current_template = template_name
+        return super().render_template(self.current_template)
+
+    def patch(self, name: str, separator: str = "\n", suffix: str = "") -> str:
+        """
+        Set the patches locations and render calls to {{ patch("...") }} from Renderer.
+        """
+        if not self.patches_locations.get(name):
+            self.patches_locations.update({name: [self.current_template]})
+        else:
+            if self.current_template not in self.patches_locations[name]:
+                self.patches_locations[name].append(self.current_template)
+
+        # Store the template's name, and replace it with the name of this patch.
+        # This handles the case where patches themselves include patches.
+        original_template = self.current_template
+        self.current_template = f"within patch: {name}"
```

Comment:
I could confirm that this works. I created the following plugin:

```
from tutor import hooks

hooks.Filters.ENV_PATCHES.add_item(
    (
        "openedx-common-settings",
        """
pouac
{{ patch('aaa-patch') }}
""",
    )
)
```

Then, `tutor config patches list` included the line:

```
aaa-patch                                       within patch: openedx-common-settings
```

Now I'm wondering if we should try to improve the formatting. "within patch" is explicit but it does not look too great in the command output. What do you think of:  `aaa-patch                    [openedx-common-settings]`?

It's a matter of personal taste of course, so if you disagree then I'm perfectly fine with your choice.

Reply:
I prefer the explicit form, but I don't have any problems with brackets; maybe we need to add somewhere the meaning of the brackets, and the issue is solved :D
What do you think about explicit vs. good-looking brackets, @kdmccormick?

Reply:
Explicit form it is!